parsl/app/futures.py

```diff
@@ -58,8 +72,17 @@ def __init__(self, fut: Future, file_obj: File, tid: Optional[int] = None) -> No
         else:
             raise ValueError("DataFuture must be initialized with a File, not {}".format(type(file_obj)))
         self.parent = fut
-
+        if app_fut:
+            self.app_fut = app_fut
+        else:
+            self.app_fut = fut
+        self.data_flow_kernel = dfk
         self.parent.add_done_callback(self.parent_callback)
+        if os.path.exists(file_obj.path):
```

Comment:
file_obj.path might not be a path on the local filesystem - in https://github.com/Parsl/parsl/pull/3712/files#diff-8ddcd7398ccac61957e74e4e419aafc09215aa3723bcb845503481bcc6edd4c9R42 there's a check that this is a `file` scheme path where that should be the case.

Reply:
fixed